dubbo-metadata-report/dubbo-metadata-definition/src/main/java/org/apache/dubbo/metadata/definition/TypeDefinitionBuilder.java

```diff
@@ -35,20 +35,18 @@
  */
 public class TypeDefinitionBuilder {
 
-    private static final ThreadLocal<ArrayList<TypeBuilder>> builders;
-
-    static {
-        builders = new ThreadLocal<ArrayList<TypeBuilder>>();
-        builders.set(new ArrayList<TypeBuilder>());
-        builders.get().add(new ArrayTypeBuilder());
-        builders.get().add(new CollectionTypeBuilder());
-        builders.get().add(new MapTypeBuilder());
-        builders.get().add(new EnumTypeBuilder());
-    }
+    private static final ThreadLocal<ArrayList<TypeBuilder>> builders = ThreadLocal.withInitial(() -> {
```

Comment:
Can we **InternalThreadLocal** instead of ThreadLocal

Reply:
Hi, because of the InternalThreadLocal depend on InternalThread.
But these method is use from spring's thread or user thread.
So I don't use InternalThreadLocal..

Reply:
ok. 

For the sake of my learning, would it be possible for explain it bit more so that I can understand the wisdom which lies in the combination of spring + threadlocal which you are using?